<a href="https://colab.research.google.com/github/tpnguyen-1999/A-B-Testing-Promotional-Banner-Strategies-for-E-commerce-Company/blob/main/%5BPython%5D_AB_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

The Ecommerce business offers a group of whitelist customers a special promotion for the membership subscription, currently they show the screen A to customers, and they found that the conversion rate of users buying the subscription is too low.\
They suggest changing the content on the banner. The suggestion as below:
(Notes: the original price of the subscription package is 199K, and with the promotion, they can buy with 99K)

Screen A: show a discounted price of paid package (99K)\
Screen B: show a discount amount in price (discount 100K)


## 1.Design AB Testing

- Hypothesis: Changing the content of the banner will change the conversion rate of users buying the subscription
- Measurable metrics: \
Primary metrics: the conversion rate of users buying the subscription
- Sample pool: All customers in whitelist, splitting the dataset equally and randomly between group 1 (screen A) and group 2 (screen B)
- External factors:\
Customer behavior: some customers may respond better to discount shown (discount 100K) rather than the price of the package (99K).
Seasonality: if the test is running during low-demand period, both screen may show lower conversion rate, etc.,


In [ ]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
path = '/content/drive/MyDrive/'
abtest = pd.read_csv(path+'abtesting.csv')
abtest.head()

Unnamed: 0  customer_id  group  is_buy
0           0     21393799      2       0
1           1     20349278      2       0
2           2     11655915      2       0
3           3     18516581      2       0
4           4      7805171      2       0

Dataset:
- customer_id: unique id of each users
- group: group1: show screenA | group2: show screenB
- is_buy: whether that user buy the subscription or not


In [ ]:
abtest.describe()

Unnamed: 0   customer_id         group        is_buy
count  30049.000000  3.004900e+04  30049.000000  30049.000000
mean   15024.000000  9.102330e+06      1.499983      0.318014
std     8674.543456  7.249900e+06      0.500008      0.465713
min        0.000000  5.500000e+01      1.000000      0.000000
25%     7512.000000  1.634794e+06      1.000000      0.000000
50%    15024.000000  7.973345e+06      1.000000      0.000000
75%    22536.000000  1.416212e+07      2.000000      1.000000
max    30048.000000  2.508438e+07      2.000000      1.000000

In [ ]:
abtest.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30049 entries, 0 to 30048
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   Unnamed: 0   30049 non-null  int64
 1   customer_id  30049 non-null  int64
 2   group        30049 non-null  int64
 3   is_buy       30049 non-null  int64
dtypes: int64(4)
memory usage: 939.2 KB


In [ ]:
group_count = abtest.groupby('group')['customer_id'].count().reset_index()
group_count

group  customer_id
0      1        15025
1      2        15024

## 2. Hypothesis Testing

In [ ]:
sample = abtest.groupby('group')['is_buy'].value_counts().reset_index()
sample

group  is_buy  count
0      1       0   9826
1      1       1   5199
2      2       0  10667
3      2       1   4357

- Hypothesis: \
Null hypothesis (H0): There is no difference in conversion rate between screen A and screen B\
Alternative hypothesis (H1): There is a difference in conversion rate between screen A and screen B
- Significance level 5%, confidence level 95%
- Two-sided

| Screen  | Amount | Conversions  |
|-------|-----|---------|
|A| 15025| 5199|
|B| 15024| 4357|

Using [AB Testing Tool](https://abtestguide.com/calc/), the test result is statistically significant, p-value = 0.0000 < 0.05 -> reject H0 and we can conclude that there is evidence to support that there is a significant difference between the conversion rate of screen A and screen B
